src/components/Router/Flow.js

```diff
@@ -0,0 +1,91 @@
+import { h, Component } from 'preact'
+import createHistory from 'history/createBrowserHistory'
+
+import { events } from '../../core'
+import {sendScreen} from '../../Tracker'
+import {wrapArray} from '../utils/array'
+
+const history = createHistory()
+
+class Flow extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      step: this.props.step,
+      componentsList: this.props.componentsList
+    }
+    this.unlisten = history.listen(({state}) => {
+      this.props.onStepChange(state)
+    })
+  }
+
+  nextStep = () => {
+    const components = this.state.componentsList
+    const currentStep = this.state.step
+    const newStepIndex = currentStep + 1
+    if (components.length === newStepIndex){
+      events.emit('complete')
+    }
+    else {
+      this.setStepIndex(newStepIndex)
+    }
+  }
+
+  previousStep = () => {
+    const currentStep = this.state.step
+    this.setStepIndex(currentStep - 1)
+  }
+
+  changeFlow = () => {
+    const flow = this.props.nextFlow()
```

Comment:
so `changeFlow` triggers a toggle between flows, correct? If so, it makes it a bit less explicit when reading the code what happens next. Did you consider `changeToFlow(flowName)` instead?